### PR TITLE
improve gff: add Locatable to Gff3BaseData, add 'Optional<String> getAttr(key)' and 'hasAttribute'

### PR DESCRIPTION
**Pull request by lindenb**
### Description

This PR improves the GFF3 classes:

- 99%  the interesting attributes in a GFF3 file are unique (for example gene_biotype, gene_name, etc...) , so using the function `List<String> getAttribute(key)` can be cumbersome. This PR adds a new function  `Optional<String> getAttr(key)` , it returns an Optional if there is zero or one value. Otherwise an exception is thrown.
- I also added a function `hasAttribute` to test for the presence of an attribute.
- add `implements Locatable` to Gff3BaseData

### Things to think about before submitting:
- [X] Make sure your changes compile and new tests pass locally.
- [X] Add new tests or update existing ones:
  - A bug fix should include a test that previously would have failed and passes now.
  - New features should come with new tests that exercise and validate the new functionality.
- [X] Extended the README / documentation, if necessary
- [X] Check your code style.
- [X] Write a clear commit title and message
  - The commit message should describe what changed and is targeted at htsjdk developers
  - Breaking changes should be mentioned in the commit message.

### CODE REVIEW

File: src/main/java/htsjdk/tribble/gff/Gff3BaseData.java [hunk 1]
```diff
@@ -152,10 +157,45 @@ public Map<String, List<String>> getAttributes() {
...
+     * Most attributes in a GFF file are present just one time in a line, e.g. : <tt>gene_biotype</tt>,  <tt>gene_name</tt>, etc ...  
+     * This function returns an <tt>Optional.empty</tt> if the <tt>key</tt> is not present,
+     *  an <tt>Optional.of(value)</tt> if there is only one value associated to the <tt>key</tt>,
+     *  or it throws an <tt>IllegalArgumentException</tt> if there is more than one value.
+     * 
+     * @param key key whose presence in the attributes is to be tested
+     * @return <tt>Optional&lt;String&gt;</tt> if this map contains zero or one attribute for the specified key
+     * @throws IllegalArgumentException if there is more than one value
+     */
+    public Optional<String> getAttr(final String key) {
```
Comment (lbergelson) [inline src/main/java/htsjdk/tribble/gff/Gff3BaseData.java:190]: I like this idea.  Can we change the name though?  Maybe getSingleAttribute()?

**Comment by lbergelson** [comment]
or getUniqueAttribute()

**Comment by lindenb** [comment]
@lbergelson thanks for the review ! I changed the name to `getUniqueAttribute`

**Review by lbergelson — APPROVED** [review]
👍

**Comment by lbergelson** [comment]
Thanks for all these additions @lindenb

**Pull request merged**